File: 1.1/src/cz/mg/vulkan/PFNvkAllocateDescriptorSets.java
package cz.mg.vulkan;

public class PFNvkAllocateDescriptorSets extends VkFunctionPointer {
    public PFNvkAllocateDescriptorSets() {
    }

    protected PFNvkAllocateDescriptorSets(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkAllocateDescriptorSets(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkAllocateDescriptorSets(long value) {
        setValue(value);
    }

    public PFNvkAllocateDescriptorSets(VkInstance instance) {
        super(instance, new VkString("vkAllocateDescriptorSets"));
    }

    public void call(VkDevice device, VkDescriptorSetAllocateInfo pAllocateInfo, VkDescriptorSet pDescriptorSets, VkResult rval){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pAllocateInfo != null ? pAllocateInfo.getVkAddress() : VkPointer.NULL, pDescriptorSets != null ? pDescriptorSets.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long device, long pAllocateInfo, long pDescriptorSets, long rval);


    public int call(VkDevice device, VkDescriptorSetAllocateInfo pAllocateInfo, VkDescriptorSet pDescriptorSets){
        return callSimplifiedNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pAllocateInfo != null ? pAllocateInfo.getVkAddress() : VkPointer.NULL, pDescriptorSets != null ? pDescriptorSets.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callSimplifiedNative(long vkaddress, long device, long pAllocateInfo, long pDescriptorSets);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkAllocateDescriptorSets.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkAllocateDescriptorSets_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pAllocateInfo, jlong pDescriptorSets){
    (void)env;
    (void)jc;
    PFN_vkAllocateDescriptorSets f = (PFN_vkAllocateDescriptorSets)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkDescriptorSetAllocateInfo*)jniLongToPointer(pAllocateInfo)),
        ((VkDescriptorSet*)jniLongToPointer(pDescriptorSets))
    );
}

